compiler/souschef/src/LexicalCast.cpp

```diff
@@ -47,7 +48,11 @@ template <> uint8_t to_number(const std::string &s)
 }
 template <> bool to_number(const std::string &s)
 {
-  if (std::stoi(s) || s == "T" || s == "t" || s == "TRUE" || s == "true")
+  if (s == "T" || s == "t" || s == "TRUE" || s == "true")
+    return true;
+  if (s == "F" || s == "f" || s == "FALSE" || s == "false")
+    return false;
+  if (std::stoi(s))
     return true;
   return false;
```

Comment:
```suggestion
  if (s == "F" || s == "f" || s == "FALSE" || s == "false" || s == "0")
    return false;
  return true;
```
How about this one?

Reply:
Anything else would be true, which seems a bit strange... like `faLse`

Reply:
Anyway... I would also like to make this simple but works good...